universal-application-tool-0.0.1/app/controllers/dev/AwsStrategy.java

```diff
@@ -0,0 +1,35 @@
+package controllers.dev;
+
+import static play.mvc.Results.redirect;
+
+import java.util.Optional;
+import models.StoredFile;
+import play.mvc.Http.Request;
+import play.mvc.Result;
+import repository.StoredFileRepository;
+
+/** Implements file uploading for AWS cloud storage. Redirects users on success. */
+public class AwsStrategy implements CloudStorageStrategy {
+
+  @Override
+  public Result create(StoredFileRepository storedFileRepository, Request request) {
+    Optional<String> etag = request.queryString("etag");
+    Optional<String> bucket = request.queryString("bucket");
+    Optional<String> key = request.queryString("key");
+    if (!bucket.isPresent() || !key.isPresent()) {
+      return redirect(routes.FileUploadController.index().url());
+    }
```

Comment:
question (blocking): is hitting this conditional an expected part of user flow or is it guarding an error case? if the latter please log an error before redirecting.